source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -286,6 +286,56 @@ void AuthenticatorImpl::handleGoodJwt(bool cache_hit) {
     }
   }
 
+  // Copy JWT Claim to Header
+  if (provider.claim_to_header_size() != 0) {
+    Json::ObjectSharedPtr payload_json;
+    try {
+      payload_json = Json::Factory::loadFromString(jwt_->payload_str_);
```

Comment:
Prefer not to do another JSON parsing again here.


jwt_verify_lib repo already parsed it into proto.Struct [here](https://github.com/google/jwt_verify_lib/blob/master/jwt_verify_lib/jwt.h#L49)

Prefer you add a couple get methods in "Jwt" class to retrieve claim fields

Reply:
Did you mean to add new getters (for object and boolean) in https://github.com/google/jwt_verify_lib/blob/master/src/jwt.cc ??

Reply:
Yes

Reply:
https://github.com/google/jwt_verify_lib/pull/74